packages/1155-contracts/test/nft/ZoraCreator1155.t.sol

```diff
@@ -1135,6 +1135,77 @@ contract ZoraCreator1155Test is Test {
         assertEq(protocolRewards.balanceOf(zora), settings.zoraReward + settings.createReferralReward);
     }
 
+    function test_SetCreatorRewardRecipientForToken() public {
+        address collaborator = makeAddr("collaborator");
+        uint256 quantity = 100;
+
+        init();
+
+        vm.prank(admin);
+        uint256 tokenId = target.setupNewToken("test", quantity);
+
+        address payable creatorRewardRecipient;
+
+        creatorRewardRecipient = target.getCreatorRewardRecipient(tokenId);
+
+        ICreatorRoyaltiesControl.RoyaltyConfiguration memory newRoyaltyConfig = ICreatorRoyaltiesControl.RoyaltyConfiguration(0, 0, collaborator);
+
+        vm.prank(admin);
+        target.updateRoyaltiesForToken(tokenId, newRoyaltyConfig);
+
+        creatorRewardRecipient = target.getCreatorRewardRecipient(tokenId);
+
+        assertEq(creatorRewardRecipient, collaborator);
+
+        vm.prank(admin);
+        target.addPermission(tokenId, address(simpleMinter), adminRole);
+
+        RewardsSettings memory settings = target.computeFreeMintRewards(quantity);
+
+        uint256 totalReward = target.computeTotalReward(quantity);
+        vm.deal(collector, totalReward);
+
+        vm.prank(collector);
+        target.mintWithRewards{value: totalReward}(simpleMinter, tokenId, quantity, abi.encode(recipient), address(0));
```

Comment:
Should we check for the rewards emit event?

Reply:
just added 👍